optimize/backend/src/test/java/io/camunda/optimize/service/util/ValidationHelperTest.java

```diff
@@ -0,0 +1,137 @@
+import io.camunda.optimize.dto.optimize.query.report.single.process.filter.ProcessFilterDto;
+import io.camunda.optimize.dto.optimize.query.report.single.process.filter.data.IdentityLinkFilterDataDto;
+import io.camunda.optimize.service.exceptions.OptimizeValidationException;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+public class ValidationHelperTest {
+
+  @Test
+  void shouldBeSuccessfulWhenDtoCorrect() {
+    // given
+    final BranchAnalysisRequestDto dto = new BranchAnalysisRequestDto();
+    dto.setFilter(
+        List.of(
+            new AssigneeFilterDto(
+                new IdentityLinkFilterDataDto(MembershipFilterOperator.IN, List.of()))));
+    dto.setEnd("3");
+    dto.setGateway("3");
+    dto.setProcessDefinitionKey("3");
+    dto.setProcessDefinitionVersion("3");
+
+    // then
+    ValidationHelper.validate(dto);
+  }
+
+  @Test
+  void shouldThrowExceptionWhenDtoIsInvalid() {
+    // given
+    final BranchAnalysisRequestDto dto = new BranchAnalysisRequestDto();
+
+    // when
+    final Throwable thrown = Assertions.catchThrowable(() -> ValidationHelper.validate(dto));
+
+    // then
+    Assertions.assertThat(thrown)
+        .isInstanceOf(OptimizeValidationException.class)
+        .hasMessageContaining("gateway activity id is not allowed to be empty or null");
+  }
+
+  @Test
+  void shouldThrowExceptionWhenTargetIsNull() {
+    // given
+    final String fieldName = "testField";
+    final Object target = null;
+
+    // when
+    final Throwable thrown =
+        Assertions.catchThrowable(() -> ValidationHelper.ensureNotEmpty(fieldName, target));
+
+    // then
+    Assertions.assertThat(thrown)
+        .isInstanceOf(OptimizeValidationException.class)
+        .hasMessageContaining("testField is not allowed to be empty or null");
+  }
+
+  @Test
+  void shouldThrowExceptionWhenCollectionIsEmpty() {
+    // given
+    final String fieldName = "testCollection";
+    final List<String> collection = Collections.emptyList();
+
+    // when
+    final Throwable thrown =
+        Assertions.catchThrowable(
+            () -> ValidationHelper.ensureCollectionNotEmpty(fieldName, collection));
+
+    // then
+    Assertions.assertThat(thrown)
+        .isInstanceOf(OptimizeValidationException.class)
+        .hasMessageContaining("testCollection is not allowed to be empty or null");
+  }
+
+  @Test
+  void shouldValidateFiltersCorrectly() {
+    // given
+    final ProcessFilterDto<?> validFilter = new InstanceStartDateFilterDto();
+    validFilter.setFilterLevel(FilterApplicationLevel.INSTANCE);
+
+    // when
+    final Throwable thrown =
+        Assertions.catchThrowable(
+            () -> ValidationHelper.validateProcessFilters(List.of(validFilter)));
+
+    // then
+    Assertions.assertThat(thrown).isInstanceOf(NullPointerException.class);
```

Comment:
this assertion does not match the test name at all. It's also testing incorrect behaviour

Reply:
fixed